module/zfs/zfs_vnops.c

```diff
@@ -2950,25 +2950,27 @@ zfs_setattr(struct inode *ip, vattr_t *vap, int flags, cred_t *cr)
 	if (mask & (ATTR_UID|ATTR_GID)) {
 
 		if (mask & ATTR_UID) {
-			SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_UID(zsb), NULL,
-			    &new_uid, sizeof (new_uid));
 			ZTOI(zp)->i_uid = SUID_TO_KUID(new_uid);
+			new_kuid = zfs_uid_read(ZTOI(zp));
+			SA_ADD_BULK_ATTR(bulk, count, SA_ZPL_UID(zsb), NULL,
+			    &new_kuid, sizeof (new_kuid));
```

Comment:
Actually the declaration is not short-lived since I'm passing the pointer to new_kuid to `SA_ADD_BULK_ATTR` and this pointer can be considered in use until the final call to `sa_bulk_update` is made. As such the lifespan of this variable is actually quite large.